### DISCUSSION TITLE
Simple ActivityPub implementation via extension (share an Eleventy website’s posts to the Fediverse)

### DISCUSSION THREAD

**Discussion by xplosionmind** [11ty/eleventy#2644] [Enhancement Queue]
Given the surge of [Mastodon](https://github.com/mastodon/mastodon)’s popularity, it is the perfect moment to focus on implementing more and more [ActivityPub](https://github.com/w3c/activitypub)-based platforms!

It would be great if Eleventy could have an extension that easily implements the ActivityPub protocol and allows to share on the Fediverse the chosen posts.

I believe the perfect library to achieve this task would be [activitypub-express](https://github.com/immers-space/activitypub-express), reportedly a “Modular ActivityPub implementation as Express JS middleware to easily add decentralization and federation to Node apps”.

## More info

- The W3C’s [ActivityPub](https://www.w3.org/TR/activitypub) specification
- [Fediverse growth stats](https://the-fediverse.info)
- xplosionmind/tommi.space#29

**Comment by yazae** [comment]
Hi

I haven’t tested it yet but there is a plugin in the work:
https://github.com/LewisDaleUK/eleventy-plugin-activity-pub

**Reply by Jolg42** [reply]
This repo now shows: "This repository was archived by the owner on Jul 3, 2025. It is now read-only."

**Comment by Jak2k** [comment]
Have a look at the IndieWeb and Bridy.
